### PR TITLE
Evochain: Owner of collection via EVM

### PR DESCRIPTION
**Pull request by dastanfv**

### CODE REVIEW

File: evolution-chain/precompile/laos-evolution/contracts/LaosEvolution.sol [hunk 1]
```diff
@@ -16,4 +16,12 @@ interface LivingAssets {
     /// @param owner the owner of the newly created collection
     /// @return collectionId the id of the newly created collection
     function createCollection(address owner) external returns (uint64);
+
+    /// @notice Owner of the collection
+    /// @dev Call this function to get the owner of a collection
+    /// @param collectionId the id of the collection
+    /// @return owner the owner of the collection
```
Comment (tonimateos) [inline evolution-chain/precompile/laos-evolution/contracts/LaosEvolution.sol:23]: If the return of the method does not have an owner, then you don't need to use it in the @return.
In this case, you should use
/// @return the owner of the collection

instead, if you had this: function ownerOfCollection(collectionId) returns (address admin);

you would have to use:
/// @return admin the owner of the collection

Comment (tonimateos) [reply evolution-chain/precompile/laos-evolution/contracts/LaosEvolution.sol:23]: And indeed, the same applies to line 17, you should just use 

    /// @return the id of the newly created collection

**Review by tonimateos — CHANGES_REQUESTED** [review]

**Review by tonimateos — APPROVED** [review]

File: evolution-chain/precompile/laos-evolution/src/lib.rs [hunk 2]
```diff
@@ -5,22 +5,23 @@ use fp_evm::{Precompile, PrecompileHandle, PrecompileOutput};
...
+	FunctionModifier, LogExt, LogsBuilder, PrecompileHandleExt,
 };
 
 use sp_core::{H160, H256};
 use sp_std::{fmt::Debug, marker::PhantomData, vec::Vec};
 
 /// Solidity selector of the CreateCollection log, which is the Keccak of the Log signature.
-pub const SELECTOR_LOG_CREATE_COLLECTION: [u8; 32] =
-	keccak256!("CreateCollection(uint256,address)");
+pub const SELECTOR_LOG_CREATE_COLLECTION: [u8; 32] = keccak256!("CreateCollection(uint64,address)");
```
[the remark this answers is not in the document]
Comment (dastanfv) [reply evolution-chain/precompile/laos-evolution/src/lib.rs]: `NewCollection` seems to be more aligned with the Solidity conventions, so I chose that

**Review by asiniscalchi — APPROVED** [review]

File: evolution-chain/precompile/laos-evolution/src/lib.rs (hunk 2 above)
[the remark this answers is not in the document]
Comment (tonimateos) [reply evolution-chain/precompile/laos-evolution/src/lib.rs]: Although I don't see explicit naming recommendations in Solidity, my experience is that normally "New" is used. I have collected some examples:

From [1]:

    // Logged when the owner of a node assigns a new owner to a subnode.
    event NewOwner(bytes32 indexed node, bytes32 indexed label, address owner);

    // Logged when the resolver for a node changes.
    event NewResolver(bytes32 indexed node, address resolver);

    // Logged when the TTL of a node changes
    event NewTTL(bytes32 indexed node, uint64 ttl);

From [2]: event NewMint


[1] https://github.com/ensdomains/ens/blob/master/contracts/ENS.sol
[2] https://basescan.org/token/0xd2b2a5d79cab06e0efc40b6c80a152a6480011a6#code

**Pull request merged**